cmd/cosign/cli/attest/attest.go

```diff
@@ -0,0 +1,228 @@
+		predicatePath = flagset.String("predicate", "", "path to the predicate file.")
+		force         = flagset.Bool("f", false, "skip warnings and confirmations")
+		idToken       = flagset.String("identity-token", "", "[EXPERIMENTAL] identity token to use for certificate from fulcio")
+		predicateType = flagset.String("type", "custom", "specify predicate type (default: custom) (slsaprovenance|link|spdx)")
+		regOpts       options.RegistryOpts
+	)
+	options.ApplyRegistryFlags(&regOpts, flagset)
+	return &ffcli.Command{
+		Name:       "attest",
+		ShortUsage: "cosign attest -key <key path>|<kms uri> [-predicate <path>] [-a key=value] [-upload=true|false] [-f] [-r] <image uri>",
+		ShortHelp:  `Attest the supplied container image.`,
+		LongHelp: `Attest the supplied container image.
+
+EXAMPLES
+  # attach an attestation to a container image Google sign-in (experimental)
+  COSIGN_EXPERIMENTAL=1 cosign attest -predicate <FILE> -type <TYPE> <IMAGE>
+
+  # attach an attestation to a container image with a local key pair file
+  cosign attest -predicate <FILE> -type <TYPE> -key cosign.key <IMAGE>
+
+  # attach an attestation to a container image with a key pair stored in Azure Key Vault
+  cosign attest -predicate <FILE> -type <TYPE> -key azurekms://[VAULT_NAME][VAULT_URI]/[KEY] <IMAGE>
+
+  # attach an attestation to a container image with a key pair stored in AWS KMS
+  cosign attest -predicate <FILE> -type <TYPE> -key awskms://[ENDPOINT]/[ID/ALIAS/ARN] <IMAGE>
+
+  # attach an attestation to a container image with a key pair stored in Google Cloud KMS
+  cosign attest -predicate <FILE> -type <TYPE> -key gcpkms://projects/[PROJECT]/locations/global/keyRings/[KEYRING]/cryptoKeys/[KEY]/versions/[VERSION] <IMAGE>
+
+  # attach an attestation to a container image with a key pair stored in Hashicorp Vault
+  cosign attest -predicate <FILE> -type <TYPE> -key hashivault://[KEY] <IMAGE>
+
+  # attach an attestation to a container image which does not fully support OCI media types
+  COSIGN_DOCKER_MEDIA_TYPES=1 cosign attest -predicate <FILE> -type <TYPE> -key cosign.key legacy-registry.example.com/my/image
+  `,
+		FlagSet: flagset,
+		Exec: func(ctx context.Context, args []string) error {
+			_ = flagset
+			_ = key
+			_ = cert
+			_ = upload
+			_ = sk
+			_ = slot
+			_ = predicatePath
+			_ = force
+			_ = idToken
+			_ = predicateType
+			_ = regOpts
+			panic("this command is now implemented in cobra.")
+		},
+	}
+}
+
+const (
+	predicateCustom = "custom"
+	predicateSlsa   = "slsaprovenance"
+	predicateSpdx   = "spdx"
+	predicateLink   = "link"
+)
+
+var predicateTypeMap = map[string]string{
+	predicateCustom: attestation.CosignCustomProvenanceV01,
+	predicateSlsa:   in_toto.PredicateSLSAProvenanceV01,
+	predicateSpdx:   in_toto.PredicateSPDX,
+	predicateLink:   in_toto.PredicateLinkV1,
+}
+
+//nolint
+func AttestCmd(ctx context.Context, ko sign.KeyOpts, regOpts options.RegistryOpts, imageRef string, certPath string,
```

Comment:
With a lot of this stuff moving into their own packages, it looks like we're getting a lot of "stutter" where the package name + function name are redundant.